faunadb-java/src/main/java/com/faunadb/client/query/Expr.java

```diff
@@ -0,0 +1,230 @@
+package com.faunadb.client.query;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.faunadb.client.types.Ref;
+import com.faunadb.client.types.SetRef;
+import com.faunadb.client.types.Value;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.Map;
+
+import static java.lang.String.format;
+
+/**
+ * A query language expression. Constructors for this class are at the {@link Language} class.
+ */
+public abstract class Expr implements Value {
```

Comment:
Ok. Let me work a little with this mindset. I'll ping you back with the outcomes.